sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIOTest.java

```diff
@@ -296,6 +297,53 @@ public void testReadWithRuntimeParametersValidationDisabled() {
     p.apply(read);
   }
 
+  @Test
+  public void testReadWithReaderStartFailed() throws IOException {
+    FailureBigtableService failureService =
+        new FailureBigtableService(
+            new AutoValue_BigtableIOTest_FailureOptions.Builder().setFailAtStart(true).build());
```

Comment:
I would advise you to use the Builder in the autovalue class, so that you're not creating the AutoValue_* class, and instead are using AutoValue builders the way they're designed: https://github.com/google/auto/blob/master/value/userguide/builders.md

```suggestion
            BigtableIOTest.builder().setFailAtStart(true).build());
```

I know this is a test file, so it's not so critical - but our users also use tests as inspiration, so let's keep a good practice here? : )

Reply:
Thanks for the instructions! It makes perfect sense.